packages/core/src/tools/limitModification.ts

```diff
@@ -34,6 +34,46 @@ export function allowModificationOfaths<T extends Context, Result>(
   return result
 }
 
+/**
+ * Current limitation:
+ * - field path do not support array, 'a.b.c' only
+ */
+export function forbidModificationOfaths<T extends Context, Result>(
+  object: T,
+  forbiddenFieldPaths: Array<FieldPath<T>>,
+  modifier: (object: T) => Result
+): Result | undefined {
+  const original = deepClone(object)
+  const result = modifier(object)
+  for (const path of forbiddenFieldPaths) {
+    set(object, path, get(original, path))
+  }
+  forbidDifferentTypes(original, object)
+  return result
+}
+
+/**
+ * Ensures that all field paths of original have the same types on target.
+ * if a path has different types then target[path] is replaced by original[path]
+ */
+function forbidDifferentTypes(original: Context, target: Context) {
```

Comment:
+1

Reply:
In addition to the types issue @BenoitZugmeyer described, I thought of the following use case:
```js
DD_LOGS.init({
  beforeSend: event => {
    if (someCondition) {
      event.foo = 'new value'
    }
  }
})
// ...
DD_LOGS.addLoggerGlobalContext({foo: 'original value'})
```
To my understanding, even if we make the `limitModification` function allow additional properties, it will not allow to edit the `foo` property (that would already exist in the event) since it's not part of the "sensitive fields". am I missing something?

Reply:
I think the function `forbidModificationOfPaths` and its usage is fine, I'm only talking about the `forbidDiffentTypes` function here.

Reply:
ah, sorry I thought we were talking about `forbidModificationOfPaths`.
In that case, yes I agree that allowing the user to change the fields however they want makes sense here.
I will remove this function.